modules/light-clients/08-wasm/keeper/keeper_test.go

```diff
@@ -58,6 +73,63 @@ func (suite *KeeperTestSuite) SetupTest() {
 	types.RegisterQueryServer(queryHelper, GetSimApp(suite.chainA).WasmClientKeeper)
 }
 
+// SetupWasmWithMockVM sets up mock cometbft chain with a mock vm.
+func (suite *KeeperTestSuite) SetupWasmWithMockVM() {
+	ibctesting.DefaultTestingAppInit = suite.setupWasmWithMockVM
+
+	suite.coordinator = ibctesting.NewCoordinator(suite.T(), 1)
+	suite.chainA = suite.coordinator.GetChain(ibctesting.GetChainID(1))
+
+	_ = storeWasmCode(suite, wasmtesting.Code)
+}
+
+func (suite *KeeperTestSuite) setupWasmWithMockVM() (ibctesting.TestingApp, map[string]json.RawMessage) {
+	suite.mockVM = wasmtesting.NewMockWasmEngine()
+	// TODO: move default functionality required for wasm client testing to the mock VM
+	suite.mockVM.StoreCodeFn = func(code wasmvm.WasmCode) (wasmvm.Checksum, error) {
+		hash := sha256.Sum256(code)
+		return wasmvm.Checksum(hash[:]), nil
+	}
+
+	suite.mockVM.PinFn = func(codeID wasmvm.Checksum) error {
+		return nil
+	}
```

Comment:
note: can be removed if #4723 lands first

Reply:
removed